examples/tutorials/advanced/grid_equalization.py

```diff
@@ -0,0 +1,222 @@
+"""
+Performing grid histogram equalization
+--------------------------------------
+The :meth:`pygmt.grdhisteq.equalize_grid` function creates a grid with
+statistics based on a cumulative distribution function.
+"""
+# sphinx_gallery_thumbnail_number = 3
+
+import pygmt
+
+###############################################################################
+# Load sample data
+# ----------------
+# Load the sample Earth relief data for a region around Yosemite valley
+# and use :meth:`pygmt.grd2xyz` to create a :class:`numpy.ndarray` with the z
+# values.
+region = [-119.825, -119.4, 37.6, 37.825]
+
+grid = pygmt.datasets.load_earth_relief(resolution="03s", region=region)
+grid_dist = pygmt.grd2xyz(grid=grid, outcols=2, output_type="numpy")[:, 0]
```

Comment:
Not directly related to this PR, but running `grd2xyz` gives me unexpected output:
```
>>> pygmt.grd2xyz(grid=grid, outcols=2, output_type="numpy")
array([[1536.,   nan,   nan],
       [1539.,   nan,   nan],
       [1542.,   nan,   nan],
       ...,
       [2793.,   nan,   nan],
       [2800.,   nan,   nan],
       [2802.,   nan,   nan]])
```
So it looks like a bug, right?

Reply:
Yes, this is a PyGMT issue related to how the temp file output is read into a pandas.DataFrame.

Reply:
Two comments here:

1. `pygmt.grd2xyz(grid=grid, outtput_type="pandas")["elevation"]` is more readable
2. Should we limit the use of `outcols` to `output_type="file"` only, otherwise we have to parse `outcols` and process the output columns.